src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -760,7 +760,10 @@ internal static bool MatchesModulePath(string modulePath, string requiredPath)
             }
 
             // Use the PowerShell filesystem provider to fully resolve the path
-            return ModuleCmdletBase.GetResolvedPath(moduleName, executionContext).TrimEnd(Path.DirectorySeparatorChar);
+            // If there is a problem, null could be returned -- so default back to the pre-normalized path
+            string normalizedPath = ModuleCmdletBase.GetResolvedPath(moduleName, executionContext) ?? moduleName;
+                
+            return normalizedPath.TrimEnd(Path.DirectorySeparatorChar);
```

Comment:
Maybe not important, but in case you get a `null` from `GetResolvedPath`, `TrimEnd` shouldn't be called on the original module name.